signer/fourbyte/abi.go

```diff
@@ -95,10 +95,10 @@ func parseSelector(selector string) ([]byte, error) {
 		Type   string    `json:"type"`
 		Inputs []fakeArg `json:"inputs"`
 	}
-	// Validate the selector and extract it's components
-	groups := selectorRegexp.FindStringSubmatch(selector)
+	// Validate the unescapedSelector and extract it's components
+	groups := selectorRegexp.FindStringSubmatch(unescapedSelector)
 	if len(groups) != 3 {
-		return nil, fmt.Errorf("invalid selector %s (%v matches)", selector, len(groups))
+		return nil, fmt.Errorf("invalid unescapedSelector %q (%v matches)", unescapedSelector, len(groups))
```

Comment:
I guess you didn't want to replace `selector` in the error string too

Reply:
doh!